app/web/src/features/dashboard/CommunityBrowser.tsx

```diff
@@ -0,0 +1,166 @@
+import CircularProgress from "components/CircularProgress";
+import {
+  useCommunity,
+  useListSubCommunities,
+} from "features/communities/hooks";
+import { LOAD_MORE } from "features/dashboard/constants";
+import { Community } from "proto/communities_pb";
+import { ReactNode, useState } from "react";
+import { Link } from "react-router-dom";
+import { routeToCommunity } from "routes";
+import makeStyles from "utils/makeStyles";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: "flex",
+    flexDirection: "row",
+    "& > * + *": {
+      marginInlineStart: theme.spacing(2),
+    },
+  },
+  list: {
+    minWidth: "10rem",
+  },
+  loader: {
+    margin: "auto",
+  },
+  selected: {
+    fontWeight: "bold",
+  },
+}));
+
+export default function CommunityBrowser() {
+  const classes = useStyles();
+  const [selected, setSelected] = useState<Community.AsObject[]>([]);
+
+  const globalCommunityQuery = useCommunity(1);
+
+  //react-query doesn't have useInfiniteQueries
+  //as a workaround, cache query results
+  //and only show "Load more" for last column
+  const query = useListSubCommunities(
+    selected?.[selected.length - 1]?.communityId || 1
+  );
+  const [cachedQueryResults, setCachedQueryResults] = useState<
+    {
+      data: Community.AsObject[];
+      hasMore?: boolean;
+    }[]
+  >([]);
+
+  const handleClick = (community: Community.AsObject, level: number) => {
+    if (level === selected.length) {
+      setCachedQueryResults([
+        ...cachedQueryResults,
+        {
+          data: query.data!.pages.flatMap((page) => page.communitiesList),
+          hasMore: query.hasNextPage,
+        },
+      ]);
+    } else {
+      setCachedQueryResults(cachedQueryResults.slice(0, level + 1));
+    }
+    setSelected([...selected.slice(0, level), community]);
+  };
+
+  return (
+    <div className={classes.root}>
+      {cachedQueryResults.map((query, index) => (
+        <BrowserColumn
+          key={index}
+          parent={selected?.[index - 1] ?? globalCommunityQuery.data}
+          communities={query.data}
+          handleClick={(community) => handleClick(community, index)}
+          selected={selected[index].communityId}
+        />
+      ))}
+      {query.isError ? (
+        <Alert severity="error">{query.error.message}</Alert>
+      ) : query.isSuccess && globalCommunityQuery.isSuccess ? (
+        <>
+          <BrowserColumn
+            parent={
+              selected?.[selected.length - 1] ?? globalCommunityQuery.data
+            }
+            communities={query.data.pages.flatMap(
+              (page) => page.communitiesList
+            )}
+            handleClick={(community) => handleClick(community, selected.length)}
+          />
+          {query.hasNextPage && (
+            <Button
+              onClick={() => query.fetchNextPage()}
+              loading={query.isFetchingNextPage}
+            >
+              {LOAD_MORE}
+            </Button>
+          )}
```

Comment:
Is this meant to appear at the next column? I'd have expect the load more button to be in the same column as the last column and below all loaded communities so far, as it looks a bit odd to have it stretched the entire column

Reply:
it's wip 😉